pkl/extensions.bzl

```diff
@@ -34,6 +34,21 @@ pkl_project = tag_class(
             allow_single_file = True,
             mandatory = True,
         ),
+        "extra_flags": attr.string_list(
+            doc = """Dictionary of name value pairs used to pass in Pkl external flags.
+                See the Pkl docs: https://pkl-lang.org/main/current/pkl-cli/index.html#command-eval""",
+            default = [],
+        ),
+        "environment": attr.string_dict(
```

Comment:
Used `environment` (instead of `env`) to better align with the terminology in [Bazel's documentation on `repository_ctx.execute`](https://bazel.build/rules/lib/builtins/repository_ctx#execute).